tests/test_errors.py

```diff
@@ -517,6 +521,137 @@ def test_all_errors():
+    assert repr(cause) == repr(ValueError('Oh no!'))
+    assert cause.__traceback__ is not None
+
+    def multi_raise_py_error(v: Any) -> Any:
+        try:
+            raise AssertionError('Wrong')
+        except AssertionError as e:
+            raise ValueError('Oh no!') from e
+
+    s2 = SchemaValidator(core_schema.no_info_plain_validator_function(multi_raise_py_error), config=enabled_config)
+    with pytest.raises(ValidationError) as exc_info:
+        s2.validate_python('anything')
+
+    def validate_s2_chain(val_err: ValidationError):
+        cause_group = val_err.__cause__
+        assert isinstance(cause_group, BaseExceptionGroup)
+        assert len(cause_group.exceptions) == 1
+
+        cause = cause_group.exceptions[0]
+        cause = check_grouped_exception(cause)
+        assert repr(cause) == repr(ValueError('Oh no!'))
+        assert cause.__traceback__ is not None
+
+        sub_cause = cause.__cause__
+        assert repr(sub_cause) == repr(AssertionError('Wrong'))
+        assert sub_cause.__cause__ is None
+        assert sub_cause.__traceback__ is not None
+
+    validate_s2_chain(exc_info.value)
+
+    # Edge case: make sure a deep inner ValidationError(s) causing a validator failure doesn't cause any problems:
+    def outer_raise_py_error(v: Any) -> Any:
+        try:
+            s2.validate_python('anything')
+        except ValidationError as e:
+            raise ValueError('Sub val failure') from e
+
+    s3 = SchemaValidator(core_schema.no_info_plain_validator_function(outer_raise_py_error), config=enabled_config)
+    with pytest.raises(ValidationError) as exc_info:
+        s3.validate_python('anything')
+
+    assert isinstance(exc_info.value.__cause__, BaseExceptionGroup)
+    assert len(exc_info.value.__cause__.exceptions) == 1
+    cause = exc_info.value.__cause__.exceptions[0]
+    cause = check_grouped_exception(cause)
+    assert repr(cause) == repr(ValueError('Sub val failure'))
+    subcause = cause.__cause__
+    assert isinstance(subcause, ValidationError)
+    validate_s2_chain(subcause)
+
+
+def test_validation_error_cause_version_variants():
+    class Result(enum.Enum):
+        CAUSE = enum.auto()
+        NO_CAUSE = enum.auto()
+        IMPORT_ERROR = enum.auto()
+
+    config: list[tuple[str, CoreConfig, Result]] = [
+        # Without the backport should still work after 3.10 as not needed:
+        (
+            'Enabled',
+            {'validation_error_cause': True},
+            Result.CAUSE if sys.version_info >= (3, 11) else Result.IMPORT_ERROR,
+        ),
+        ('Disabled specifically', {'validation_error_cause': False}, Result.NO_CAUSE),
+        ('Disabled implicitly', {}, Result.NO_CAUSE),
+    ]
+
+    # Simulate the package being missing:
+    with patch.dict('sys.modules', {'exceptiongroup': None}):
+
+        def singular_raise_py_error(v: Any) -> Any:
+            raise ValueError('Oh no!')
+
+        for _, config, expected in config:
```

Comment:
please use parameterize rather than a loop - loops in tests are impossible to understand/debug.

Reply:
Done!